src/java.base/linux/classes/jdk/internal/platform/cgroupv1/CgroupV1Subsystem.java

```diff
@@ -0,0 +1,490 @@
+    }
+
+    public int[] getEffectiveCpuSetMems() {
+        return CgroupSubsystemController.stringRangeToIntArray(CgroupSubsystemController.getStringValue(cpuset, "cpuset.effective_mems"));
+    }
+
+    public double getCpuSetMemoryPressure() {
+        return CgroupV1SubsystemController.getDoubleValue(cpuset, "cpuset.memory_pressure");
+    }
+
+    public Boolean isCpuSetMemoryPressureEnabled() {
+        long val = getLongValue(cpuset, "cpuset.memory_pressure_enabled");
+        return (val == 1);
+    }
+
+
+    /*****************************************************************
+     * Memory Subsystem
+     ****************************************************************/
+
+
+    public long getMemoryFailCount() {
+        return getLongValue(memory, "memory.failcnt");
+    }
+
+    public long getMemoryLimit() {
+        long retval = getLongValue(memory, "memory.limit_in_bytes");
+        if (retval > CgroupV1SubsystemController.UNLIMITED_MIN) {
+            if (memory.isHierarchical()) {
+                // memory.limit_in_bytes returned unlimited, attempt
+                // hierarchical memory limit
+                String match = "hierarchical_memory_limit";
+                retval = CgroupV1SubsystemController.getLongValueMatchingLine(memory,
+                                                            "memory.stat",
+                                                            match);
+            }
+        }
+        return CgroupV1SubsystemController.longValOrUnlimited(retval);
+    }
+
+    public long getMemoryMaxUsage() {
+        return getLongValue(memory, "memory.max_usage_in_bytes");
+    }
+
+    public long getMemoryUsage() {
+        return getLongValue(memory, "memory.usage_in_bytes");
+    }
+
+    public long getKernelMemoryFailCount() {
+        return getLongValue(memory, "memory.kmem.failcnt");
+    }
+
+    public long getKernelMemoryLimit() {
+        return CgroupV1SubsystemController.longValOrUnlimited(getLongValue(memory, "memory.kmem.limit_in_bytes"));
+    }
+
+    public long getKernelMemoryMaxUsage() {
+        return getLongValue(memory, "memory.kmem.max_usage_in_bytes");
+    }
+
+    public long getKernelMemoryUsage() {
+        return getLongValue(memory, "memory.kmem.usage_in_bytes");
+    }
+
+    public long getTcpMemoryFailCount() {
+        return getLongValue(memory, "memory.kmem.tcp.failcnt");
+    }
+
+    public long getTcpMemoryLimit() {
+        return CgroupV1SubsystemController.longValOrUnlimited(getLongValue(memory, "memory.kmem.tcp.limit_in_bytes"));
+    }
+
+    public long getTcpMemoryMaxUsage() {
+        return getLongValue(memory, "memory.kmem.tcp.max_usage_in_bytes");
+    }
+
+    public long getTcpMemoryUsage() {
+        return getLongValue(memory, "memory.kmem.tcp.usage_in_bytes");
+    }
+
+    public long getMemoryAndSwapFailCount() {
+        return getLongValue(memory, "memory.memsw.failcnt");
+    }
```

Comment:
This is missing the hunk for https://bugs.openjdk.java.net/browse/JDK-8250984. See:
https://github.com/openjdk/jdk11u-dev/blob/87703322e980e42035d7cf101f04cafdc31fe278/src/java.base/linux/classes/jdk/internal/platform/cgroupv1/Metrics.java#L480..L482